packages/sockethub-platform-xmpp/src/index.js

```diff
@@ -311,7 +311,7 @@ class XMPP {
       const status = job.object.content || '';
       // setting presence
       this.debug('setting presence: ' + show + ' status: ' + status);
-      this.__client.send(xml(show, { type: status })).then(done);
+      this.__client.send(xml("presence", { show }, { status })).then(done);
```

Comment:
According to you explanation in https://github.com/sockethub/sockethub/pull/676#issuecomment-1030861554, doesn't this then never set presence for a `groupchat` as per spec? What does "show" actually mean here?

Reply:
Looks like `show`, if set, can only be one of `dnd`, `away`, `chat`, `xa` ... if not set, client assumed to be available. https://xmpp.org/rfcs/rfc3921.html#stanzas-presence-children-show

So the code needs to be fixed more. I'm not sure what enum we should give the `presence` property in the long run (eg. do we just match the XMPP list?), but in the short run we can just say if it's `available` don't set the `show` property at all, and if it's anything else, set `away` ?

Reply:
OK, I dug into it a bit further. I am still not 100% sure I understand how XMPP handles presence, because there's the presence `type` property and the `show` child element, both of which handle some sort of presence. https://xmpp.org/rfcs/rfc3921.html#stanzas-presence-children-show

I think I got a basic working version defined now, including cleaning up the presence object schema as it was slightly incorrect. Added some more tests, etc. Can you have a look, specifically at this commit, and let me know what you think? https://github.com/sockethub/sockethub/pull/676/commits/65eb793dfd74b9e823ac340edd497d3e31ce7b8a